### PR TITLE
make k3d registry port configurable via a flag

### PR DESCRIPTION
**Pull request by dennis-ge**
**Description**

Changes proposed in this pull request:

- quick fix for https://github.com/kyma-project/cli/issues/1061

**Related issue(s)**
Fixes https://github.com/kyma-project/cli/issues/1061

### CODE REVIEW

File: internal/k3d/registry.go [hunk 1]
```diff
@@ -22,3 +35,62 @@ func (cl *RegistryList) Unmarshal(data []byte) error {
 	cl.Registries = registries
 	return nil
 }
+
+type Config struct {
+	MappedRegistryPort string `yaml:"mapped_registry_port"`
+}
+
+func SaveRegistryPort(port string) error {
```
Comment (chrkl) [inline internal/k3d/registry.go:43]: Please add a short Godoc style comment about the purpose of this function (here and below)

**Review by chrkl — COMMENTED** [review]
Deploying a serverless function unfortunately does not work. To test this run
```
kyma init function
kyma apply function
```

The builder pod cannot push the image to the registry:
```
error checking push permissions -- make sure you entered the correct tag name, and that you are authenticated correctly, and try again: checking push permission for "k3d-kyma-registry:5001/default-function-loving-krzysztof5:3b076b3ef5b4f765e87e8a8d4bf6522812d9c75768db40379a88eae9b9c8710e": creating push check transport for k3d-kyma-registry:5001 failed: Get "https://k3d-kyma-registry:5001/v2/": dial tcp 172.24.0.2:5001: connect: connection refused; Get "http://k3d-kyma-registry:5001/v2/": dial tcp 172.24.0.2:5001: connect: connection refused
```

The the registry container still has a port forwarding to 5000 in Docker. I am wondering if that causes the problem:
```
f58332ff3b79   registry:2                  "/entrypoint.sh /etc…"   12 minutes ago   Up 12 minutes   0.0.0.0:5001->5000/tcp                                              k3d-kyma-registry
```

**Review by chrkl — APPROVED** [review]

File: docs/gen-docs/kyma_alpha_provision_k3d.md [hunk 2]
```diff
@@ -20,6 +20,7 @@ kyma alpha provision k3d [flags]
   -k, --kube-version string    Kubernetes version of the cluster (default "1.20.11")
       --name string            Name of the Kyma cluster (default "kyma")
   -p, --port strings           Map ports 80 and 443 of K3D loadbalancer (e.g. -p 80:80@loadbalancer -p 443:443@loadbalancer) (default [80:80@loadbalancer,443:443@loadbalancer])
+      --registry-port string   Specify the port on which the k3d registry will be exposed. (default "5001")
```
Comment (mmitoraj) [inline docs/gen-docs/kyma_alpha_provision_k3d.md:23]:
```suggestion
      --registry-port string   Specify the port on which the k3d registry will be exposed (default "5001")
```

**Review by mmitoraj — COMMENTED** [review]
I removed the full stop to be consistent with the other entries in the document. Thank you.

**Review by mmitoraj — APPROVED** [review]

**Review by chrkl — APPROVED** [review]

**Pull request merged**